### PR TITLE
Add e2e tests for feature NodeLogQuery

### PR DESCRIPTION
**Pull request by fabi200123**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Adds e2e tests for the alpha feature NodeLogQuery.

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

- [KEP]: KEP 2258: [KEP 2258: add node log query](https://github.com/kubernetes/kubernetes/pull/96120)
- [Enhancements]: [Node log query](https://github.com/kubernetes/enhancements/issues/2258)
- [Documentation]: [KEP-2258: Node log query documentation](https://github.com/kubernetes/website/pull/40003)

### CODE REVIEW

File: test/e2e/node/kubelet.go [hunk 1]
```diff
@@ -450,4 +453,236 @@ var _ = SIGDescribe("kubelet", func() {
...
+				cmd := tk.KubectlCmd("get", "--raw", queryCommand)
+				result := runKubectlCommand(cmd, "sinceTime")
+				logs := journalctlCommand("-u", "kubelet", "--utc")
+				assertContains(result, logs)
+			}
+		})
+	})
 })
+
+func runKubectlCommand(cmd *exec.Cmd, arg string) (result string) {
```
Comment (aravindhp) [inline test/e2e/node/kubelet.go:648]: Do we need the arg input just for printing purposes? I suggest printing out `cmd.Args` instead.

File: test/e2e/node/kubelet.go [hunk 2]
```diff
@@ -450,4 +453,236 @@ var _ = SIGDescribe("kubelet", func() {
 			}
 		})
 	})
+
+	// Tests for NodeLogQuery feature
```
Comment (aravindhp) [inline test/e2e/node/kubelet.go:457]: We will need to figure out how to enable the feature gate and the kubelet option to enable this feature for the tests.

**Review by aravindhp — COMMENTED** [review]
Thanks for working on this, @fabi200123. I have left a few comments.

File: test/e2e/node/kubelet.go [hunk 3]
```diff
@@ -450,4 +453,236 @@ var _ = SIGDescribe("kubelet", func() {
...
+	ginkgo.Describe("kubectl get --raw \"/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=/<insert-log-file-name-here> [Feature:NodeLogQuery]", func() {
+		const (
+			RFC3339 = "2006-01-02T15:04:05+00:00"
```
[the remark this answers is not in the document]
Comment (fabi200123) [reply test/e2e/node/kubelet.go]: Thank you, modified it

File: test/e2e/node/kubelet.go (hunk 1 above)
Comment (fabi200123) [reply test/e2e/node/kubelet.go]: Modified it

**Comment by aravindhp** [comment]
/test pull-kubernetes-e2e-capz-windows

Previous run [failed](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/117011/pull-kubernetes-e2e-capz-windows/1673733373807300608) on:
```   
Summarizing 1 Failure:
    [FAIL] [sig-windows] [Feature:WindowsHostProcessContainers] [MinimumKubeletVersion:1.22] HostProcess containers [It] container command path validation
    test/e2e/windows/host_process.go:449
```

**Review by aravindhp — COMMENTED** [review]
/lgtm

From a feature author perspective. We can get this merged and follow it up with this running as part of the Windows e2e tests. We need to figure out how to add the kubelet option as part of that.

**Comment by rphillips** [comment]
lgtm

/assign @mrunalp

**Review by mrunalp — APPROVED** [review]

**Pull request merged**